### PR TITLE
Increase test coverage to 93% #549

### LINKED ISSUE #549: Increase test coverage to 93%

### PR DESCRIPTION
**Pull request by fgimenez**
Fixes #549

# 🚀 Description

Adds missing test cases, aiming to 93% coverage. The local coverage output seems to be different from coverall's, let's see if more changes are required to reach 93% :)

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](/docs/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [ ] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

**Comment by fgimenez** [comment]
Already >93! :tada: 

Ready for review.

File: test/crowdsale/AllowanceCrowdsale.test.js [hunk 1]
```diff
@@ -66,3 +68,13 @@ contract('AllowanceCrowdsale', function ([_, investor, wallet, purchaser, tokenW
...
 });
+
+contract('AllowanceCrowdsaleFail', function ([_, investor, wallet, purchaser, tokenWallet]) {
+  const rate = new BigNumber(1);
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  it('should fail with token wallet different from token address', async function () {
+    this.token = await SimpleToken.new({ from: tokenWallet });
+    await AllowanceCrowdsaleFail.new(rate, wallet, this.token.address, ZERO_ADDRESS).should.be.rejectedWith(EVMRevert);
+  });
```
Comment (facuspagnuolo) [inline test/crowdsale/AllowanceCrowdsale.test.js]: Two things here:
- Can we move this scenario to a `describe` block within the `contract` block above?
- What about using `await assertRevert(...)` ?

File: test/Heritable.test.js [hunk 2]
```diff
@@ -63,6 +67,19 @@ contract('Heritable', function (accounts) {
...
+    const someRandomAddress = accounts[2];
+    await expectThrow(heritable.proclaimDeath({ from: owner }));
+    await expectThrow(heritable.proclaimDeath({ from: someRandomAddress }));
+  });
+
+  it('heir can\'t proclaim death if owner is death', async function () {
+    const heir = accounts[1];
+    await heritable.setHeir(heir, { from: owner });
+    heritable.proclaimDeath({ from: heir });
+    await expectThrow(heritable.proclaimDeath({ from: heir }));
```
Comment (facuspagnuolo) [inline test/Heritable.test.js]: can we use `assertRevert` here?

File: test/examples/SampleCrowdsale.test.js [hunk 3]
```diff
@@ -112,3 +113,31 @@ contract('SampleCrowdsale', function ([owner, wallet, investor]) {
...
+    this.closingTime = this.openingTime + duration.weeks(1);
+    this.afterClosingTime = this.closingTime + duration.seconds(1);
+
+    this.token = await SampleCrowdsaleToken.new({ from: owner });
+    this.vault = await RefundVault.new(wallet, { from: owner });
+
+    await SampleCrowdsaleFail.new(
+      this.openingTime, this.closingTime, RATE, wallet, CAP, this.token.address, GOAL
+    ).should.be.rejectedWith(EVMRevert);
+  });
```
Comment (facuspagnuolo) [inline test/examples/SampleCrowdsale.test.js]: same here, can we show different scenarios using different `describe` blocks within the same `contract` block?

File: test/crowdsale/AllowanceCrowdsale.test.js (hunk 1 above)
Comment (fgimenez) [reply test/crowdsale/AllowanceCrowdsale.test.js]: Absolutely, much better now, no need for `AllowanceCrowdsaleFail` either.

File: test/Heritable.test.js (hunk 2 above)
Comment (fgimenez) [reply test/Heritable.test.js]: Sure, I've taken the opportunity to use it in all the PR changes.

File: test/examples/SampleCrowdsale.test.js (hunk 3 above)
Comment (fgimenez) [reply test/examples/SampleCrowdsale.test.js]: Sure, done.

**Comment by fgimenez** [comment]
Thanks a lot for your review @facuspagnuolo :) PTAL.

File: test/examples/SampleCrowdsale.test.js [hunk 4]
```diff
@@ -111,4 +114,12 @@ contract('SampleCrowdsale', function ([owner, wallet, investor]) {
     const balanceAfterRefund = web3.eth.getBalance(investor);
     balanceBeforeInvestment.should.be.bignumber.equal(balanceAfterRefund);
   });
+
+  describe('when goal > cap', function () {
```
Comment (facuspagnuolo) [inline test/examples/SampleCrowdsale.test.js]: nit: we can move the `HIGH_GOAL` variable to this scope

Comment (fgimenez) [reply test/examples/SampleCrowdsale.test.js]: Good point, done.

**Comment by fgimenez** [comment]
Sorry, the last build passed https://travis-ci.org/OpenZeppelin/zeppelin-solidity/builds/362965719, you mean the error in the previous one https://travis-ci.org/OpenZeppelin/zeppelin-solidity/builds/362964989 ?

File: test/Heritable.test.js [hunk 5]
```diff
@@ -63,6 +68,19 @@ contract('Heritable', function (accounts) {
...
+  it('only heir can proclaim death', async function () {
+    const someRandomAddress = accounts[2];
+    await assertRevert(heritable.proclaimDeath({ from: owner }));
+    await assertRevert(heritable.proclaimDeath({ from: someRandomAddress }));
+  });
+
+  it('heir can\'t proclaim death if owner is death', async function () {
+    const heir = accounts[1];
+    await heritable.setHeir(heir, { from: owner });
+    heritable.proclaimDeath({ from: heir });
```
[the remark this answers is not in the document]
Comment (fgimenez) [reply test/Heritable.test.js]: thx for your review @frangio, sure, fixed.

**Comment by frangio** [comment]
Thank you @fgimenez! :tada:

**Pull request merged**